archunit/src/test/java/com/tngtech/archunit/core/domain/DependencyTest.java

```diff
@@ -71,6 +71,20 @@ public void Dependency_from_throws_declaration() {
                 .contains("Method <" + origin.getFullName() + "> throws type <" + IOException.class.getName() + ">");
     }
 
+    @Test
+    public void Dependency_from_instanceof_check() {
+        JavaMethod origin = importClassesWithContext(ClassWithDependencyOnInstanceofCheck.class, SomeMemberType.class)
+                .get(ClassWithDependencyOnInstanceofCheck.class).getMethod("method", Object.class);
+        InstanceofCheck instanceofCheck = getOnlyElement(origin.getInstanceofChecks());
+
+        Dependency dependency = Dependency.tryCreateFromInstanceofCheck(instanceofCheck).get();
+
+        assertThat(dependency.getOriginClass()).matches(ClassWithDependencyOnInstanceofCheck.class);
+        assertThat(dependency.getTargetClass()).matches(SomeMemberType.class);
+        assertThat(dependency.getDescription()).as("description")
+                .contains("Method <" + origin.getFullName() + "> checks instanceof <" + SomeMemberType.class.getName() + ">");
```

Comment:
If you add the line number to `InstanceofCheck` you can add a `SourceCodeLocation` to `Dependency` (compare method calls, etc.). Then you could assert the correct location link snippet here (`Foo.java:123`)